src/Saml/SamlProvider.php

```diff
@@ -91,6 +93,71 @@ private function hydrateUser(User $user, SamlLoginAttributes $token): void
             }
         }
 
+        $teamAttribute = $this->configuration->getTeamsAttribute();
+        $teamMapping = $this->configuration->getTeamsMapping();
+
+        // extract user teams from a special saml attribute
+        if (!empty($teamAttribute) && $token->hasAttribute($teamAttribute)) {
+            $teamMap = [];
+            foreach ($teamMapping as $mapping) {
+                $field = $mapping['kimai'];
+                $team = $this->teamRepository->findById($field)[0];
```

Comment:
I do not like repository magic methods. Please change that to:
```suggestion
                $team = $this->teamRepository->findOneBy(['id' => $field]);
                if ($team !== null) {
                ...
```